src/Entities/Player.cpp

```diff
@@ -1702,6 +1702,23 @@ void cPlayer::SendRotation(double a_YawDegrees, double a_PitchDegrees)
 
 
 
+void cPlayer::SpectateEntity(cEntity * a_Target)
+{
+	if (static_cast<cPlayer *>(a_Target) == this)
```

Comment:
Is the cast necessary? If so, I would prefer upcasting `this` (guarunteed to be valid) over downcasting `a_Target` (which might not be a `cPlayer`).